### PR TITLE
Upgrade lottie-react-native to 6.3.1

### PR DESCRIPTION
**Pull request by ZhenjaHorbach**
### Details

Upgrade E/App to use lottie-react-native 6.3.1+

### Fixed Issues

$ https://github.com/Expensify/App/issues/28132
PROPOSAL: https://github.com/Expensify/App/issues/28132#issuecomment-1733655247

### Tests

1. Open the app 
2. Moving pictures are displayed correctly in the app. For example: SignIn image, Workspaces Header image and Preferences Header image in settings

- [x] Verify that no errors appear in the JS console

### Offline tests

1. Open the app 
2. Moving pictures are displayed correctly in the app. For example: SignIn image, Workspaces Header image and Preferences Header image in settings

### QA Steps

1. Open the app 
2. Moving pictures are displayed correctly in the app. For example: SignIn image, Workspaces Header image and Preferences Header image in settings

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/68128028/a3e3b4f4-0f76-4037-9b09-138afbb27268

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/68128028/213f24f2-4ab6-4277-8b3a-4fd7f5921737

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/68128028/af97025c-3204-4bd4-ac4e-4d14964ff5c4

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/68128028/2f8eb82b-2a15-44a7-9e65-cc0db5920928

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/68128028/74e4b7c4-1a8f-4bbd-9197-2a7aa6bf9146

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/68128028/15999ee8-b75a-45e1-966d-c2f832672bfe

</details>

### CODE REVIEW

**Comment by dangrous** [comment]
@roryabraham do you want to review this? Happy to do so but looks like it's your issue

**Comment by roryabraham** [comment]
Sure, sorry for the delay. Going to run a test build to make this faster to test

File: patches/lottie-react-native+6.3.1.patch [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
++    let sizeStyle;
++    if (
++      !styleObject ||
++      (styleObject.width === undefined && styleObject.height === undefined)
++    ) {
++      sizeStyle =
++        autoSize && sources.sourceJson
++          ? { width: (source as any).w }
++          : StyleSheet.absoluteFill;
++    }
```
Comment (kosmydel) [inline patches/lottie-react-native+6.3.1.patch:56]: I don't think that we want to simply revert [the breaking changes](https://github.com/lottie-react-native/lottie-react-native/pull/992/files#diff-989a0a34ad1a1c41bcd903ca888ab92e813b132c99db023f860eb6bddc7a736a) in the patch.

**Review by kosmydel — CHANGES_REQUESTED** [review]
We should address the breaking changes in another way than just reverting them.

**Comment by dangrous** [comment]
I'm OOO this week so unassigning myself as a reviewer, I think y'all should probably have it covered!

File: patches/lottie-react-native+6.3.1.patch (hunk 1 above)
Comment (situchan) [reply patches/lottie-react-native+6.3.1.patch:56]: Can we override using props? I don't really like patching

Comment (kosmydel) [reply patches/lottie-react-native+6.3.1.patch:56]: We might also consider passing `aspectRatio` as a value/prop instead of calculating it. Just an idea.

Comment (kosmydel) [reply patches/lottie-react-native+6.3.1.patch:56]: We should probably also put it into some `useMemo`. I think that this is still much better than doing it in a patch. Especially when I am working on [this related issue](https://github.com/Expensify/App/issues/26871). And patching would require regular maintenance if the original library changes.

---

But **there is a problem** with this approach. When we move to [the `.lottie`](https://github.com/Expensify/App/issues/26871) files we can't access the `.w` and `.h` so easily, as these animations won't be JSON objects anymore. I'm not sure how should we handle it.

---

cc @situchan @roryabraham  what do you think?

File: src/components/ConfirmationPage.js [hunk 2]
```diff
@@ -46,7 +46,7 @@ function ConfirmationPage(props) {
                     source={props.animation}
                     autoPlay
                     loop
-                    style={styles.confirmationAnimation}
+                    style={[styles.confirmationAnimation, styles.aspectRatioLottie(props.animation)]}
```
Comment (kosmydel) [inline src/components/ConfirmationPage.js:49]: We could possibly add some kind of wrapper to the `<Lottie>` component, adding this style, instead of copying the same line several times, with the argument, that is already passed to the component.
Just an idea.

File: package-lock.json [hunk 3]
```diff
@@ -80371,14 +80365,10 @@
...
-      "peer": true
-    },
     "lottie-react-native": {
-      "version": "5.1.6",
-      "resolved": "https://registry.npmjs.org/lottie-react-native/-/lottie-react-native-5.1.6.tgz",
-      "integrity": "sha512-vhdeZstXMfuVKwnddYWjJgQ/1whGL58IJEJu/iSf0XQ5gAb4pp/+vy91mdYQLezlb8Aw4Vu3fKnqErJL2hwchg==",
+      "version": "6.3.1",
+      "resolved": "https://registry.npmjs.org/lottie-react-native/-/lottie-react-native-6.3.1.tgz",
+      "integrity": "sha512-M18nAVYeGMF//bhL27D2zuMcrFPH0jbD/deBvcWi0CCcfZf6LQfx45xt+cuDqwr5nh6dMm+ta8KfZJmkbNhtlg==",
       "requires": {
```
Comment (situchan) [inline package-lock.json:80372]: Not sure why this file I generated is different from this branch.

Replace package-lock.json with this done:

[package-lock.json.zip](https://github.com/Expensify/App/files/12840510/package-lock.json.zip)

File: ios/Podfile.lock [hunk 4]
```diff
@@ -188,33 +188,33 @@ PODS:
     - GoogleUtilities/Network (~> 7.4)
```
Comment (situchan) [inline ios/Podfile.lock:188]: We only need to update lottie related frameworks. Revert all other changes which are out of scope.

Just replace with this file I generated:

[Podfile.lock.zip](https://github.com/Expensify/App/files/12840514/Podfile.lock.zip)

**Review by situchan — CHANGES_REQUESTED** [review]
Please pull main

File: src/styles/styles.js [hunk 5]
```diff
@@ -3779,6 +3779,13 @@ const styles = (theme) => ({
         lineHeight: variables.lineHeightLarge,
     },
 
+    aspectRatioLottie: (source) => {
+        if (typeof source === 'object' && !source.uri) {
```
Comment (situchan) [inline src/styles/styles.js:3783]: We can add more safety condition here, though all Lottie JSON may include `w`, `h` properties.
```suggestion
        if (!source.uri && typeof source === 'object' && source.w && source.h) {
```

File: src/styles/styles.js [hunk 6]
```diff
@@ -3779,6 +3779,13 @@ const styles = (theme) => ({
...
+            return {aspectRatio: source.w / source.h};
+        }
+        return {aspectRatio: '1'};
```
Comment (situchan) [inline src/styles/styles.js:3786]: I prefer setting this undefined and let user fully customize style properties.
If both width and height are defined, they're skipped because aspectRatio overwrites them.
```suggestion
        return {};
```

**Review by situchan — COMMENTED** [review]
There are 8 lottie animations in the app. Let's make sure all of them are tested in all platforms.

**Comment by situchan** [comment]
Check this video which demonstrates all animations are working well. You should test like this on all platforms.


https://github.com/Expensify/App/assets/108292595/01e987d7-93d4-4cc0-a263-1e06c2053d07

**Comment by situchan** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>


https://github.com/Expensify/App/assets/108292595/178f7d45-d529-4f0a-bc03-d8517e4ea84a



</details>

<details>
<summary>Mobile Web - Chrome</summary>


https://github.com/Expensify/App/assets/108292595/a927c8c6-14e7-4030-95f9-a5bb025c7722



</details>

<details>
<summary>Mobile Web - Safari</summary>


https://github.com/Expensify/App/assets/108292595/8ece14ce-d20f-4005-9667-15fcd1e0aef2



</details>

<details>
<summary>Desktop</summary>


https://github.com/Expensify/App/assets/108292595/46c7f0e3-234a-44d2-bc0f-c958bbf36075



</details>

<details>
<summary>iOS</summary>


https://github.com/Expensify/App/assets/108292595/58cdf82e-4d8b-4535-88d6-b720fce389ac



</details>

<details>
<summary>Android</summary>


https://github.com/Expensify/App/assets/108292595/defde901-bdb4-447a-a54a-e3cfbff860af



</details>

**Review by situchan — APPROVED** [review]
LGTM
@roryabraham all yours

**Comment by mountiny** [comment]
@ZhenjaHorbach can you please resolve the conflicts?

**Review by mountiny — APPROVED** [review]
@situchan has kindly retested IOS and its working, I will move this ahead

File: src/components/Lottie/index.js [hunk 7]
```diff
@@ -0,0 +1,3 @@
+import Lottie from './Lottie';
```
Comment (blazejkustra) [inline src/components/Lottie/index.js:1]: Why `Lottie.tsx` is written in TS and `index.js` in JS? I think Lottie catalog can be removed and `Lottie.tsx` file in `components` catalog should be sufficient.

@ZhenjaHorbach

Comment (situchan) [reply src/components/Lottie/index.js:1]: > Why `Lottie.tsx` is written in TS and `index.js` in JS?

This is already existing pattern, i.e. MapView but agree to write TS for all

> I think Lottie catalog can be removed and `Lottie.tsx` file in `components` catalog should be sufficient.

Agree. Unless platform specific or has propTypes in js
